### ISSUE TITLE
IDCNN测试的时候同一个输入会有不同的输出结果

### ISSUE THREAD

**Issue by Scagin** [crownpku/Information-Extraction-Chinese#59]
我想问下，为什么在测试的时候，同一个输入每一次的输出结果都不一样。我看过最后输出的分数矩阵，也是每次不一样的，为什么呢？我看过源代码，也看不出来是什么原因，tensorflow的静态计算图又不利于调试，所以我想上来寻找答案。
I would like to ask why the output of the same input is not the same at the time of the test. I have seen the final output of the score matrix, which is not the same every time. Why? I have read the source code and I can't see why it is. tensorflow's static calculations are not conducive to debugging, so I come here for answers.

**Comment by michael92ht** [comment]
@Scagin  我现在也遇到了 你解决这个问题了吗

**Comment by NealHappy** [comment]
@Scagin @michael92ht  我也遇到了这个问题，有解决方案吗

**Comment by Scagin** [comment]
@michael92ht @NealHappy 当时我希望能通过tensorboard进行调试，查看网络层的中间参数的，但是确实tensorflow的静态计算图对调试非常不友好，最终我也没有得出结论，所以放弃了。希望你们能有更好的方法能得出正确的结论，然后在这里分享一下，十分感谢！

**Comment by YijinHuang** [comment]
在model.py的def IDCNN_layer里，作者是通过if self.dropout == 1.0来判断是否是训练状态，但是self.dropout是tensor对象，该判断恒为False，因此测试的时候，对输入进行了dropout导致了结果不一样。只要在创造Model对象的时候另外传入一个boolean参数替代以上的判断，就可以解决这个问题了。

**Comment by logye** [comment]
从现象看，问题定位在
def IDCNN_layer(self, model_inputs, name=None):
一种修改方法：
dropout_keep 设置为1.0
`        reuse = False
        if self.config["dropout_keep"] == 1.0:   # self.dropout 替换为 self.config["dropout_keep"]
            reuse = True
`
另一种方法：
注释掉 finalOut = tf.nn.dropout(finalOut, keepProb)
实测训练时，在IDCNN_layer 不进行dropout，反而较原有各性能指标有提升，综合接近BiLSTM （ 仅在我的训练集上是如此表现，未做深入分析

**Comment by Scagin** [comment]
@logye
非也，从feed_dict来看，预测的时候并没有使用到dropout层。当初我和你有相同的猜想，并且把drop_prob改为1.0后，输出仍然是不稳定的。

**Comment by Scagin** [comment]
@YijinHuang
原来如此，我没有注意到self.dropout==1.0恒为false，非常感谢。
